storage/innobase/handler/i_s.cc

```diff
@@ -5765,9 +5792,17 @@ i_s_sys_columns_fill_table(
 		mutex_exit(&dict_sys.mutex);
 
 		if (!err_msg) {
-			i_s_dict_fill_sys_columns(thd, table_id, col_name,
-						 &column_rec, nth_v_col,
-						 tables->table);
+			err = i_s_dict_fill_sys_columns(
+				thd, table_id, col_name,
+				&column_rec, nth_v_col,
+				tables->table);
+			if (err) {
+				ut_free(pcur.old_rec_buf);
```

Comment:
I think that the `ut_free()` had better be invoked after `func_exit:`, to fix yet another memory leak.